ksl.py

```diff
@@ -31,7 +46,9 @@ class KSL(object):
     def __init__(self):
         self.thread_pool = concurrent.futures.ThreadPoolExecutor(max_workers=4)
 
-    def __do_request(self, args):
+    @staticmethod
```

Comment:
I don't see a reason for this to be decorated a `staticmethod`, it's not called outside the class.

Reply:
A method being static or non-static doesn't have any bearing on whether or not it's called externally. A method being static or non-static depends on whether or not it accesses or mutates the state of object. If a class method doesn't rely on any object state (`self`), then it should be static.